### PR TITLE
[WIP] Add RGB <=> YCbCr conversion.

### PR DESCRIPTION
**Pull request by S-aiueo32**

### CODE REVIEW

File: kornia/color/ycbcr.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            type(image)))
+
+    if len(image.shape) < 3 or image.shape[-3] != 3:
+        raise ValueError("Input size must have a shape of (*, 3, H, W).Got {}"
+                         .format(image.shape))
+
+    r: torch.Tensor = image[..., 0, :, :]
+    g: torch.Tensor = image[..., 1, :, :]
+    b: torch.Tensor = image[..., 2, :, :]
+
```
Comment (edgarriba) [inline kornia/color/ycbcr.py]: I would reference somehow from where did you get those weights. As a first search I found wiki page: https://es.wikipedia.org/wiki/YCbCr

File: kornia/color/ycbcr.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    y: torch.Tensor = 0.29900 * r + 0.58700 * g + 0.11400 * b
+    cb: torch.Tensor = -0.16874 * r - 0.33126 * g + 0.50000 * b + 128
```
Comment (edgarriba) [inline kornia/color/ycbcr.py]: 0.16874 -> 0.168736 ; 0.33126 -> 0.331264

File: kornia/color/ycbcr.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    cr: torch.Tensor = 0.50000 * r - 0.41869 * g - 0.08131 * b + 128
```
Comment (edgarriba) [inline kornia/color/ycbcr.py]: 0.41869 -> 0.418688 ; 0.08131 -> 0.081312

File: kornia/color/ycbcr.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
+import torch
+import torch.nn as nn
+
+
+class RgbToYcbcr(torch.nn.Module):
+    r"""Convert image from RGB to YCbCr.
+    The image data is assumed to be in the range of (0, 1).
+
+    args:
+        image (torch.Tensor): RGB image to be converted to YCbCr
```
Comment (edgarriba) [inline kornia/color/ycbcr.py]: add endpoint `RGB iRGB image to be converted to YCbCr.`

File: test/color/test_ycbcr.py [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+                             [[0., 255., 0.],
+                              [0., 255., 0.],
+                              [0., 255., 0.]],
+
+                             [[0., 0., 255.],
+                              [0., 0., 255.],
+                              [0., 0., 255.]]]) / 255
+
+        expected = torch.tensor([[[76, 149, 29],
```
Comment (edgarriba) [inline test/color/test_ycbcr.py]: need to be in floating point
`torch.tensor([[[76., 149., 29.],`

File: test/color/test_ycbcr.py [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
```
Comment (edgarriba) [inline test/color/test_ycbcr.py]: floating point

File: test/color/test_ycbcr.py [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+                                 [[0., 255., 0.],
+                                  [0., 255., 0.],
+                                  [0., 255., 0.]],
+
+                                 [[0., 0., 255.],
+                                  [0., 0., 255.],
+                                  [0., 0., 255.]]]) / 255
+
+        data = torch.tensor([[[76, 149, 29],
```
Comment (edgarriba) [inline test/color/test_ycbcr.py]: floating point

File: test/color/test_ycbcr.py [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
```
Comment (edgarriba) [inline test/color/test_ycbcr.py]: floating point

File: kornia/color/ycbcr.py (hunk 1 above)
Comment (S-aiueo32) [reply kornia/color/ycbcr.py]: Google returns https://www.wdic.org/w/WDIC/YCbCr on the top in Japanese, which provides wrong info.

**Comment by S-aiueo32** [comment]
I fixed all you said.
However, I'm sure that mypy still returns errors, because the build logs indicate that the errors is in the codes I don't modified.

**Comment by S-aiueo32** [comment]
I noticed why it doesn't pass.
I wrote static tests in value range [0, 1], but there is adding 128 for cb and cr channel, my bad.
Is there any manner to specify value range?

**Comment by edgarriba** [comment]
@S-aiueo32 if the range values are well stated in the docs (which I think they are) it shouldn't be a problem. For consistency can we keep it in 0/1 ?

**Comment by edgarriba** [comment]
@S-aiueo32 from where did you get the expected results ? did you compute by hand or from opencv ?

**Comment by S-aiueo32** [comment]
From opencv.
```
data = np.array([[[255, 0, 0],
                  [255, 0, 0],
                  [255, 0, 0]],

                 [[0, 255, 0],
                  [0, 255, 0],
                  [0, 255, 0]],

                 [[0, 0, 255],
                  [0, 0, 255],
                  [0, 0, 255]]], dtype='uint8')
data = np.transpose(data, [1, 2, 0])

cv2_ycbcr = cv2.cvtColor(data, cv2.COLOR_RGB2YCrCb)
cr = cv2_ycbcr[:,:,1].copy()
cb = cv2_ycbcr[:,:,2].copy()
cv2_ycbcr[:,:,1], cv2_ycbcr[:,:,2] = cb, cr
cv2_ycbcr = np.transpose(cv2_ycbcr, [2, 0, 1])
```

**Comment by S-aiueo32** [comment]
@edgarriba @ducha-aiki 
Thank you for reviewing.

I'm sorry about not doing anything about this PR.
The rejection from CI is caused by rounding errors.
I have to edit the test script to pass all checks but I don't know the acceptable error range you want.
Any suggestion?

**Comment by edgarriba** [comment]
@S-aiueo32 I will take a look today and will merge it. What other features do you have in mind to implement. Maybe we can help.

**Comment by edgarriba** [comment]
@S-aiueo32 I did a small refactor and a quick test. From where did you get the test data/expected ? I see that there are several implementations of this algorithm in different libraries (opencv, pil, etc)

@ducha-aiki any suggestions ?

**Comment by edgarriba** [comment]
@ananyamukh6 could you take a look at this ?

**Comment by edgarriba** [comment]
closing this in favor of https://github.com/kornia/kornia/pull/345

**Pull request closed without merging**